### PR TITLE
Add permissions page for Kubernetes Agent

### PR DESCRIPTION
**Pull request by scme0**
[[sc-76944]](https://app.shortcut.com/octopusdeploy/story/76944/documentation-permissions)

# Target Namespaces Screen Shot
localhost_3000_docs_infrastructure_deployment-targets_kubernetes_kubernetes-agent_permissions (5)

# Cluster Role Rules Screen Shot
localhost_3000_docs_infrastructure_deployment-targets_kubernetes_kubernetes-agent_permissions (6)

### CODE REVIEW

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+description: Information about what permissions are required and how to adjust them
+navOrder: 10
+---
+
+The Kubernetes agent uses service accounts to manage access to cluster objects.
+
+There are 3 main components that run with different permissions in the Kubernetes agent:
+- **Agent Pod** - This is the main component and is responsible for receiving work from Octopus Server and scheduling it in the cluster.
+- **Script Pods** - These are run to execute work on the cluster. When Octopus issues work to the agent, the Tentacle will schedule a pod to run the script to execute the required work. These are short-lived, single-use pods which are removed by Tentacle when they are complete.
+- **NFS Server Pod** - This optional component is used if no Storage Class is specified during installation.
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md:15]:
```suggestion
- **NFS Server Pod** - This optional component is used if no StorageClass is specified during installation.
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+--set tentacle.targetName="Nonproduction Agent" \
+--set tentacle.serverUrl="http://localhost:5000/" \
+--set tentacle.serverCommsAddress="http://localhost:10943/" \
+--set tentacle.space="Default" \
+--set tentacle.targetEnvironments="{Development,Preproduction}" \
+--set tentacle.targetRoles="{k8s-cluster-tag}" \
+--set tentacle.bearerToken="XXXX" \
+--version "1.*.*" \
+--create-namespace --namespace octopus-agent-my-cluster \
+my-cluster \
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md:55]:
```suggestion
--set agent.acceptEula="Y" \
--set agent.targetName="Nonproduction Agent" \
--set agent.serverUrl="http://localhost:5000/" \
--set agent.serverCommsAddress="http://localhost:10943/" \
--set agent.space="Default" \
--set agent.targetEnvironments="{Development,Preproduction}" \
--set agent.targetRoles="{k8s-cluster-tag}" \
--set agent.bearerToken="XXXX" \
--version "1.*.*" \
--create-namespace --namespace octopus-agent-my-agent \
my-agent\
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          - 'services'
+          verbs:
+          - '*'
+        - nonResourceURLs:
+          - '*'
+          verbs:
+          - '*'
+
+tentacle:
+  ACCEPT_EULA: 'Y'
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md:81]:
```suggestion
agent:
  acceptEula: 'Y'
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    - 'k8s-cluster-tag'
+  bearerToken: 'XXXX'
+```
+**script:**
+```Bash
+helm upgrade --install --atomic \
+--values values.yaml \
+--version "1.*.*" \
+--create-namespace --namespace octopus-agent-my-cluster \
+my-cluster \
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md:98]:
```suggestion
--create-namespace --namespace octopus-agent-my-agent\
my-agent \
```

File: src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+oci://registry-1.docker.io/octopusdeploy/kubernetes-agent
+```
+
+
+# NFS Server Pod Permissions
+
+If you have not provided a storageClassName for persistence, an NFS pod will be used. This NFS Server pod requires `privileged` access. For more information see [Kubernetes agent Storage](/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage#nfs-storage).
```
Comment (APErebus) [inline src/pages/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/permissions/index.md:105]:
```suggestion
If you have not provided a predefined storageClassName for persistence, an NFS pod will be used. This NFS Server pod requires `privileged` access. For more information see [Kubernetes agent Storage](/docs/infrastructure/deployment-targets/kubernetes/kubernetes-agent/storage#nfs-storage).
```

**Review by APErebus — APPROVED** [review]
Some minor language, but looks excellent!

**Review by zentron — APPROVED** [review]
This looks good. The content seems really helpful.

It might look a little cleaner condensing the different options for injecting values in a tabbed form e.g. (https://octopus.com/docs/projects/variables/output-variables, https://github.com/OctopusDeploy/docs/blob/8b95759c323f695414450318574e840144d4b9be/src/pages/docs/projects/variables/output-variables.md?plain=1#L2)

Listing the 3 configuration options once like that is probably ok, but if we end up showing these different ways to configure the chart in multiple places it may be cleaner with the control.  @nikita-dergilev or @kirstenschwarzer  might have thoughts.

**Comment by scme0** [comment]
@zentron do you mean the examples? like the scripts for `scriptPods.serviceAccount.targetNamespaces` and `scriptPods.serviceAccount.clusterRole.Rules`?

**Comment by scme0** [comment]
@zentron Like this?
# scriptPods.serviceAccount.targetNamespaces Screen Shot
localhost_3000_docs_infrastructure_deployment-targets_kubernetes_kubernetes-agent_permissions (2)

# scriptPods.serviceAccount.clusterRole.Rules Screen Shot
localhost_3000_docs_infrastructure_deployment-targets_kubernetes_kubernetes-agent_permissions (3)

**Comment by scme0** [comment]
@zentron I've merged this with the tabs that I think you were after as I think it looks pretty good but if we need to tweak a few things we can in a new PR 👍

**Pull request merged**